src/ledctl/ledctl.c

```diff
@@ -555,73 +779,109 @@ static led_status_t _cmdline_ibpi_parse(int argc, char *argv[])
 static led_status_t _cmdline_parse_non_root(int argc, char *argv[])
 {
 	int opt_index, opt = -1;
-	led_status_t status = LED_STATUS_SUCCESS;
+	char *shortopts;
+	struct option *longopts;
 
-	do {
-		opt = getopt_long(argc, argv, shortopt, longopt, &opt_index);
-		switch (opt) {
-		case 'v':
+	optind = test_params ? optind : 1;
+
+	setup_options(&longopts, &shortopts, possible_params, ARRAY_SIZE(possible_params));
```

Comment:
we don't need all options I think

Reply:
We can pass all options here, below we check only non-root ones. We can also create new list "non-root parameters". 2 solutions are valid.